sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -248,11 +256,15 @@ export function convertResponseForMetrics(
   });
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars -- eslint doesn't recognize that the extracted variables are prefixed with '_' and are purposefully unused.
-  const { resourceregion, value: _ignoredValue, interval, ...rest } = generatedResponse;
+  const { resourceregion, value: _ignoredValue, interval, timespan, ...rest } = generatedResponse;
 
   const obj: MetricsQueryResult = {
     ...rest,
-    metrics
+    metrics,
+    timespan: convertIntervalToTimeIntervalObject(timespan),
+    getMetricByName(metricName) {
+      return this.metrics.find((it) => it.name === metricName);
+    }
```

Comment:
I think you could optimize this by defining the function `getMetricByName` outside of this function, and just assigned it to the `MetricsQueryResult` object.

Currently, you're creating a new function for every metric result. Changing how this is built would allow you to re-use a single function.

Rough example:
```ts
function getMetricByName(metricName: string) {
  return this.metrics.find((it) => it.name === metricName);
}

function createMetricsQueryResult(...params): MetricsQueryResult {
  return { ...params, getMetricByName };
}
```

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/17544